symengine/series_piranha.cpp

```diff
@@ -155,6 +201,35 @@ Expression UPSeriesPiranha::convert(const Number &x) {
     return Expression(x.rcp_from_this());
 }
 
+RCP<const Basic> UPSeriesPiranha::as_basic() const {
+    RCP<const Symbol> x = symbol(var_);
+    umap_basic_num dict_;
+    for (const auto& it : p_) {
+        if (it.first != 0) {
+            const RCP<const Basic> &term = SymEngine::mul(SymEngine::pow(x, SymEngine::integer(it. second.degree())), it.first.get_basic());
+            RCP<const Number> coef;
+            RCP<const Basic> t;
+            Add::as_coef_term(term, outArg(coef), outArg(t));
+            Add::dict_add_term(dict_, coef, t);
```

Comment:
`term` can be an `Add` for `x**0`. `dict_add_term` doesn't handle that. `coef_dict_add_term` does